tools/ci_build/github/azure-pipelines/templates/jobs/win-ci-vs-2022-job.yml

```diff
@@ -221,12 +218,14 @@ jobs:
         workingDirectory: '$(Build.BinariesDirectory)\${{ parameters.BuildConfig }}\${{ parameters.BuildConfig }}'
         displayName: 'Run tests'
 
-      - task: PublishTestResults@2
+      - powershell: |
+          $testResultsFiles = Get-ChildItem -Path $(Build.BinariesDirectory)\${{ parameters.BuildConfig }} -Recurse -Filter *.results.xml
+          $testResultsFilesPath = $testResultsFiles.FullName -join ";"
+          
+          Write-Host "Test results files: $testResultsFilesPath"
+          
+          Write-Host "##vso[results.publish type=JUnit;mergeResults=true;publishRunAttachments=true;resultFiles=$testResultsFilesPath;]Unit Test Run"
```

Comment:
does the PublishTestResults task not work?
is there documentation for this `##vso[results.publish ...]` logging command? I didn't find it.

Reply:
It failed because with the same symlink error when deleting intermediate files,

This is the docs for the ##vso 

https://learn.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash

Reply:
I did see that doc page, but it doesn't mention `results.publish`.

if this is an undocumented command, it may be better to use the official PublishTestResults task. if that task has trouble traversing some directories maybe we can first get the list of results files (as you've done here) and then provide them to the task directly.

Reply:
I don't think it is documented, I found it being used in some Tests in the sdk:
https://github.com/microsoft/azure-pipelines-tasks/blob/f8d35bd6dc6a9c7e371753775af0d5d375f90030/Tasks/PublishTestResultsV1/PublishTestResults.ps1#L116

I can modify it to use PublishTask and list the files using powershell.

Reply:
This step is optional. If it caused much trouble for us, we can remove it.

Reply:
> This step is optional. If it caused much trouble for us, we can remove it.

I think it should be good now.

Reply:
hm, maybe it's not quite working yet:
https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1320580&view=logs&j=0cbe10c0-f8dc-5be9-4d18-f0451d1e6fe2&t=7592abce-6c8d-510f-a306-2fa321189c7d

```
##[warning]No test result files matching '[
  'onnxruntime_customopregistration_test.exe.Debug.results.xml,onnxruntime_global_thread_pools_test.exe.Debug.results.xml,onnxruntime_logging_apis_test.exe.Debug.results.xml,onnxruntime_shared_lib_test.exe.Debug.results.xml,onnxruntime_test_all.exe.Debug.results.xml'
]' were found.
```

Reply:
we can get rid of it as it is optional then.